pkg/observable/notifiable/observable_test.go

```diff
@@ -0,0 +1,162 @@
+package notifiable_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"pocket/pkg/observable"
+	"pocket/pkg/observable/notifiable"
+)
+
+const (
+	notifyTimeout            = 100 * time.Millisecond
+	unsubscribeSleepDuration = notifyTimeout * 2
+)
+
+func TestNewNotifiableObservable(t *testing.T) {
+	type test struct {
+		name     string
+		notifier chan int
+	}
+
+	input := 123
+	nonEmptyBufferedNotifier := make(chan int, 1)
+	nonEmptyBufferedNotifier <- input
+
+	tests := []test{
+		{name: "nil notifier", notifier: nil},
+		{name: "empty non-buffered notifier", notifier: make(chan int)},
+		{name: "empty buffered len 1 notifier", notifier: make(chan int, 1)},
+		{name: "non-empty buffered notifier", notifier: nonEmptyBufferedNotifier},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			notifee, notifier := notifiable.NewObservable[int](tt.notifier)
+			require.NotNil(t, notifee)
+			require.NotNil(t, notifier)
+
+			// construct 3 distinct subscriptions, each with its own channel
+			subscriptions := make([]observable.Subscription[int], 3)
+			for i := range subscriptions {
+				subscriptions[i] = notifee.Subscribe(ctx)
+			}
+
+			group := errgroup.Group{}
+			notifiedOrTimedOut := func(subscriptionCh <-chan int) func() error {
```

Comment:
OPTIONAL: consider passing in a subscription struct instead of the underlying channel.